roles/ceph-osd/tasks/crush_rules.yml

```diff
@@ -2,29 +2,29 @@
 - name: configure crush hierarchy
   ceph_crush:
     cluster: "{{ cluster }}"
-    location: "{{ hostvars[item]['osd_crush_location'] }}"
-    containerized: "{{ container_exec_cmd }}"
-  with_items: "{{ groups[osd_group_name] }}"
+    location: "{{ osd_crush_location }}"
+    containerized: "{{ hostvars[groups[mon_group_name][0]]['container_exec_cmd'] | default('') }}"
   register: config_crush_hierarchy
+  delegate_to: '{{ groups[mon_group_name][0] }}'
   when:
-    - inventory_hostname == groups.get(mon_group_name) | last
-    - create_crush_tree | bool
-    - hostvars[item]['osd_crush_location'] is defined
+    - hostvars[groups[mon_group_name][0]]['create_crush_tree'] | default(false) | bool
+    - osd_crush_location is defined
 
 - name: create configured crush rules
-  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} osd crush rule create-simple {{ item.name }} {{ item.root }} {{ item.type }}"
-  with_items: "{{ crush_rules | unique }}"
+  command: "{{ hostvars[groups[mon_group_name][0]]['container_exec_cmd'] | default('') }} ceph --cluster {{ cluster }} osd crush rule {{ 'create-replicated' if item.class is defined else 'create-simple' }} {{ item.name }} {{ item.root }} {{ item.type }} {{ item.class | default('') }}"
```

Comment:
We probably need to be sure that the device class exists to avoid failures

Reply:
that would add more complexity but I agree we should go this way

Reply:
In fact this solution only moves the issue later.
If we `list the existing device class + register` and use `when + in register.stdout` then the crush rule won't be created if the device class doesn't exist.
At the end we will have the issue on the pool creation task because the crush rule won't exist.